File: investment/backend/model/stock.go
package model

// AccountType represents the type of investment account.
type AccountType string

const (
	AccountNISA     AccountType = "NISA"
	AccountIDeCo    AccountType = "iDeCo"
	AccountSpecific AccountType = "特定"
)

// Stock represents a single stock holding in the portfolio.
type Stock struct {
	Code         string      `json:"code"`
	Name         string      `json:"name"`
	Quantity     int         `json:"quantity"`
	AcquirePrice float64     `json:"acquire_price"`
	AcquireCost  float64     `json:"acquire_cost"`
	CurrentPrice float64     `json:"current_price"`
	MarketValue  float64     `json:"market_value"`
	ProfitLoss   float64     `json:"profit_loss"`
	ProfitRate   float64     `json:"profit_rate"`
	AccountType  AccountType `json:"account_type"`
}

// PortfolioResponse is the API response for GET /api/portfolio.
type PortfolioResponse struct {
	Stocks []Stock `json:"stocks"`
	Total  Summary `json:"total"`
}

// Summary holds aggregated portfolio values.
type Summary struct {
	TotalAcquireCost float64 `json:"total_acquire_cost"`
	TotalMarketValue float64 `json:"total_market_value"`
	TotalProfitLoss  float64 `json:"total_profit_loss"`
	TotalProfitRate  float64 `json:"total_profit_rate"`
}

File: investment/frontend/src/types/stock.ts
export type AccountType = "NISA" | "iDeCo" | "特定";

export interface Stock {
  code: string;
  name: string;
  quantity: number;
  acquire_price: number;
  acquire_cost: number;
  current_price: number;
  market_value: number;
  profit_loss: number;
  profit_rate: number;
  account_type: AccountType;
}

export interface Summary {
  total_acquire_cost: number;
  total_market_value: number;
  total_profit_loss: number;
  total_profit_rate: number;
}

export interface PortfolioResponse {
  stocks: Stock[];
  total: Summary;
}

export interface ScreeningStock {
  code: string;
  name: string;
  price: number;
  per: number;
  pbr: number;
  dividend_yield: number;
  market_cap: number;
  industry: string;
}

export interface ScreeningFilter {
  per_min?: number;
  per_max?: number;
  pbr_min?: number;
  pbr_max?: number;
  dividend_yield_min?: number;
  dividend_yield_max?: number;
  market_cap_min?: number;
  market_cap_max?: number;
  industry?: string;
}

export interface ScreeningResponse {
  stocks: ScreeningStock[];
  count: number;
}
